test/e2e/network/network_policy.go

```diff
@@ -1890,69 +1890,67 @@ var _ = SIGDescribe("NetworkPolicy [Feature:SCTPConnectivity][LinuxOnly][Disrupt
 })
 
 func testCanConnect(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int) {
-	testCanConnectProtocol(f, ns, podName, service, targetPort, v1.ProtocolTCP)
+	testConnectProtocol(f, ns, podName, service, targetPort, v1.ProtocolTCP,  false)
 }
 
 func testCannotConnect(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int) {
-	testCannotConnectProtocol(f, ns, podName, service, targetPort, v1.ProtocolTCP)
+	testConnectProtocol(f, ns, podName, service, targetPort, v1.ProtocolTCP, true)
 }
 
 func testCanConnectProtocol(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int, protocol v1.Protocol) {
-	ginkgo.By(fmt.Sprintf("Creating client pod %s that should successfully connect to %s.", podName, service.Name))
-	podClient := createNetworkClientPod(f, ns, podName, service, targetPort, protocol)
-	defer func() {
-		ginkgo.By(fmt.Sprintf("Cleaning up the pod %s", podClient.Name))
-		if err := f.ClientSet.CoreV1().Pods(ns.Name).Delete(context.TODO(), podClient.Name, metav1.DeleteOptions{}); err != nil {
-			framework.Failf("unable to cleanup pod %v: %v", podClient.Name, err)
-		}
-	}()
-	checkConnectivity(f, ns, podClient, service)
+	testConnectProtocol(f, ns, podName, service, targetPort, protocol, false)
 }
 
 func testCannotConnectProtocol(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int, protocol v1.Protocol) {
-	ginkgo.By(fmt.Sprintf("Creating client pod %s that should not be able to connect to %s.", podName, service.Name))
-	podClient := createNetworkClientPod(f, ns, podName, service, targetPort, protocol)
+	testConnectProtocol(f, ns, podName, service, targetPort, protocol,  true)
+}
+func testConnectProtocol(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int, protocol v1.Protocol, expectErr bool) {
+	tryCount := 3
+	pods := make([]*v1.Pod, 0, tryCount)
+	var conErr error
+	var pod *v1.Pod
+	connString := "should"
+	if expectErr {
+		connString = "should not"
+	}
+
+	// Network policy can take some time to program, especially in scaled/busy environments. Let's allow some retries to
+	// account for that, until we have status field available on network policy object
+	for i := 0; i < tryCount; i++ {
+		ginkgo.By(fmt.Sprintf("Creating client pod %s that %s successfully connect to %s.", podName, connString, service.Name))
```

Comment:
nit: Consider calling the `By()` method be called outside the `tryCount` loop?

Reply:
yeah, that is a little cleaner. Done.